gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java

```diff
@@ -354,12 +354,20 @@
       new HadoopConfigurationProperty<>(
           "fs.gs.grpc.server.address", GoogleCloudStorageOptions.DEFAULT_GCS_GRPC_SERVER_ADDRESS);
 
+  /** Configuration key for check interval (in millisecond) for gRPC request timeout to GCS. */
+  public static final HadoopConfigurationProperty<Long> GCS_GRPC_CHECK_INTERVAL_TIMEOUT_MS =
+      new HadoopConfigurationProperty<>("fs.gs.grpc.checkinterval.timeout.ms", 1_000L);
+
   /**
    * Configuration key for the connection timeout (in millisecond) for gRPC read requests to GCS.
    */
   public static final HadoopConfigurationProperty<Long> GCS_GRPC_READ_TIMEOUT_MS =
       new HadoopConfigurationProperty<>("fs.gs.grpc.read.timeout.ms", 30 * 1000L);
 
+  /** Configuration key for the message timeout (in millisecond) for gRPC read requests to GCS. */
+  public static final HadoopConfigurationProperty<Long> GCS_GRPC_READ_MESSAGE_TIMEOUT_MS =
```

Comment:
should we have a `grpc.message_level_timeout.enabled`? to turn on/off this feature? or maybe allow default value to 0 which means don't wait on this this (read/write) message level timeout

Reply:
The value of 0 would turn the feature off

Reply:
oh.. that's nice. We should document it then saying that setting it to 0 will disable the timeout.